plugins/helm/parser/src/templateanalyzer.cpp

```diff
@@ -0,0 +1,603 @@
+  {
+    secretTemplate.depends = -1;
+    secretTemplate.id = createIdentifier(secretTemplate);
+  }
+  else
+  {
+    secretTemplate.depends = serviceIter->serviceId;
+    secretTemplate.id = createIdentifier(secretTemplate);
+    addEdge(service_.serviceId, secretTemplate.depends, secretTemplate.id, secretTemplate.kind);
+  }
+
+  addHelmTemplate(secretTemplate);
+}
+
+void TemplateAnalyzer::processResources(
+  const std::string& path_,
+  YAML::Node& currentFile_,
+  model::Microservice& service_)
+{
+  /* If the resource keys are not found, return. */
+  YAML::Node resourcesKey = findKey("resources", currentFile_);
+  if (!resourcesKey.IsDefined())
+    return;
+
+  YAML::Node requestsKey = findKey("requests", resourcesKey);
+  if (!requestsKey.IsDefined())
+    return;
+
+  /* Collect the resources. */
+
+  for (const auto& pair : requestsKey)
+  {
+    auto it = _msResourcePairs.find(YAML::Dump(pair.first));
+    if (it == _msResourcePairs.end())
+      return;
+
+    model::MSResource resource;
+    resource.type = it->second;
+    resource.service = service_.serviceId;
+
+    auto convertedAmount = convertUnit(YAML::Dump(pair.second), it->second);
+    resource.amount = convertedAmount.first;
+    resource.unit = convertedAmount.second;
+
+    _msResources.push_back(resource);
+  }
+}
+
+void TemplateAnalyzer::processStorageResources(
+  const std::string& path_,
+  YAML::Node& currentFile_,
+  model::Microservice& service_)
+{
+  /* --- Collect storage resources --- */
+  YAML::Node volumeClaimsKey = findKey("volumeClaimTemplates", currentFile_);
+  if (!volumeClaimsKey.IsDefined())
+    return;
+
+  YAML::Node storageKey = findKey("storage", volumeClaimsKey);
+  if (!storageKey.IsDefined())
+  {
+    LOG(info) << " storageKey not found";
+    return;
+  }
+
+  auto it = _msResourcePairs.find("storage");
+  if (it == _msResourcePairs.end())
+    return;
+
+  model::MSResource resource;
+  resource.type = it->second;
+  resource.service = service_.serviceId;
+
+  LOG(info) << YAML::Dump(storageKey);
+  auto convertedAmount = convertUnit(YAML::Dump(storageKey), it->second);
+  resource.amount = convertedAmount.first;
+  resource.unit = convertedAmount.second;
+
+  _msResources.push_back(resource);
+}
+
+std::pair<float, std::string> TemplateAnalyzer::convertUnit(
+  std::string amount_,
+  model::MSResource::ResourceType type_)
+{
+  switch (type_)
+  {
+    case model::MSResource::ResourceType::CPU:
+    {
+      int m = amount_.find('m');
+      if (m != std::string::npos)
+      {
+        amount_.erase(m, 1);
+        return std::make_pair(std::stof(amount_) / 1000, "");
+      }
+      else
+      {
+        return std::make_pair(std::stof(amount_), "");
+      }
+    }
+
+    case model::MSResource::ResourceType::MEMORY:
+    case model::MSResource::ResourceType::STORAGE:
+    {
+      int m = amount_.find('M');
+      if (m != std::string::npos)
+      {
+        amount_.erase(m, 2);
+        LOG(info) << std::stof(amount_);
```

Comment:
This and other log commands produces some strange and verbose output, e.g.:

> 2023-08-10 15:44:30 [INFO] 50Gi
> 2023-08-10 15:44:30 [INFO] 50
> 2023-08-10 15:44:30 [INFO] 256
> 2023-08-10 15:44:30 [INFO] 8Gi
> 2023-08-10 15:44:30 [INFO] 8
> 2023-08-10 15:44:30 [INFO] 8Gi
> 2023-08-10 15:44:30 [INFO] 8

There should be more context and the messages either debug or trace level.